disco_aws_automation/disco_vpc.py

```diff
@@ -266,45 +268,113 @@ def delete_instance_routes(self, instance):
                                    RouteTableId=route_table.id,
                                    DestinationCidrBlock=route.destination_cidr_block)
 
-    def _configure_dhcp(self):
+    def _get_ntp_server_config(self):
+        ntp_servers = []
+        ntp_server_config = self.get_config("ntp_server")
+        if ntp_server_config:
+            for ntp_server in ntp_server_config.split():
+                if self._is_valid_ip(ntp_server):
+                    ntp_servers.append(ntp_server)
+                else:
+                    ntp_server_ip = socket.gethostbyname(ntp_server)
+                    ntp_servers.append(ntp_server_ip)
+        else:
+            ntp_server_metanetwork = self.get_config("ntp_server_metanetwork")
+            ntp_server_offset = self.get_config("ntp_server_offset")
+            if ntp_server_metanetwork and ntp_server_offset:
+                ntp_servers.append(str(
+                    self.networks[ntp_server_metanetwork].ip_by_offset(ntp_server_offset)))
+
+        return ntp_servers if ntp_servers else None
+
+    def _is_valid_ip(self, ip_str):
+        try:
+            IPAddress(ip_str)
+            return True
+        except AddrFormatError:
+            return False
+
+    def _update_dhcp_options(self, dry_run=False):
+        desired_dhcp_options = self._get_dhcp_configs()
+        logger.info("Desired DHCP options: %s", desired_dhcp_options)
+
+        try:
+            existing_dhcp_options = throttled_call(
+                self.boto3_ec2.describe_dhcp_options,
+                DhcpOptionsIds=[self.vpc['DhcpOptionsId']],
+                Filters=[{'Name': 'tag:Name', 'Values': [self.environment_name]}]
+            )['DhcpOptions'][0]['DhcpConfigurations']
+
+        except IndexError:
+            existing_dhcp_options = []
+
+        for option in existing_dhcp_options:
+            option['Values'] = [value['Value'] for value in option['Values']]
+
+        logger.info("Existing DHCP options: %s", existing_dhcp_options)
+
+        if not dry_run and not self._same_dhcp_options(desired_dhcp_options, existing_dhcp_options):
```

Comment:
any reason to skip update step when they are same rather than just overwrite always?